sunpy/coordinates/transformations.py

```diff
@@ -174,17 +174,23 @@ def wrapped_func(*args, **kwargs):
 
 
 def _observers_are_equal(obs_1, obs_2):
-    # Note that this also lets pass the situation where both observers are None
-    if obs_1 == obs_2:
+    # Handle None cases
+    if obs_1 is None and obs_2 is None:
         return True
-
-    # obs_1 != obs_2
     if obs_1 is None:
         raise ConvertError("The source observer is set to None, but the destination observer is "
                            f"{obs_2}.")
     if obs_2 is None:
         raise ConvertError("The destination observer is set to None, but the source observer is "
                            f"{obs_2}.")
+    # Make sure obs_1 and obs_2 are in the same frame
+    if not np.all(obs_1.is_equivalent_frame(obs_2)):
+        return False
+
+    if np.all(obs_1 == obs_2):
```

Comment:
This now returns a boolean array if there are multiple coordinates, hence the `np.all` call.